pkg/network/setup/configstate_test.go

```diff
@@ -118,36 +125,85 @@ var _ = Describe("config state", func() {
 		injectedErr := fmt.Errorf("fail discovery")
 		discover, config := &funcStub{errRun: injectedErr}, &funcStub{}
 
-		Expect(configState.Run(testNet, discover.f, config.f)).To(MatchError(injectedErr))
+		Expect(configState.Run(nics, discover.f, config.f)).To(MatchError(injectedErr))
 
-		Expect(discover.executed).To(BeTrue(), "the discover step should execute")
-		Expect(config.executed).To(BeFalse(), "the config step should not execute")
+		Expect(discover.executedNICs).To(Equal([]string{testNet0}), "the discover step should execute")
+		Expect(config.executedNICs).To(BeEmpty(), "the config step should not execute")
 
-		_, err := cache.ReadPodInterfaceCache(&baseCacheCreator, uid, testNet)
+		_, err := cache.ReadPodInterfaceCache(&baseCacheCreator, uid, testNet0)
 		Expect(errors.Is(err, os.ErrNotExist)).To(BeTrue(), "expect no cache to exist")
 	})
 
 	It("runs and fails at the config step", func() {
 		injectedErr := fmt.Errorf("fail config")
 		discover, config := &funcStub{}, &funcStub{errRun: injectedErr}
 
-		Expect(configState.Run(testNet, discover.f, config.f)).To(MatchError(injectedErr))
+		Expect(configState.Run(nics, discover.f, config.f)).To(MatchError(injectedErr))
 
-		Expect(discover.executed).To(BeTrue(), "the discover step should execute")
-		Expect(config.executed).To(BeTrue(), "the config step should execute")
+		Expect(discover.executedNICs).To(Equal([]string{testNet0}), "the discover step should execute")
+		Expect(config.executedNICs).To(Equal([]string{testNet0}), "the config step should execute")
 
-		podIFaceCacheDataResult, err := cache.ReadPodInterfaceCache(&baseCacheCreator, uid, testNet)
+		podIFaceCacheDataResult, err := cache.ReadPodInterfaceCache(&baseCacheCreator, uid, testNet0)
 		Expect(err).NotTo(HaveOccurred())
 		Expect(podIFaceCacheDataResult.State).To(Equal(cache.PodIfaceNetworkPreparationStarted))
 	})
+
+	When("with multiple interfaces", func() {
+		BeforeEach(func() {
+			nics = append(nics,
+				podNIC{podInterfaceName: testNet1},
+				podNIC{podInterfaceName: testNet2},
+			)
+		})
+
+		It("runs with no current state (cache is empty)", func() {
+			discover, config := &funcStub{}, &funcStub{}
+
+			Expect(configState.Run(nics, discover.f, config.f)).To(Succeed())
+
+			Expect(discover.executedNICs).To(Equal([]string{testNet0, testNet1, testNet2}))
+			Expect(config.executedNICs).To(Equal([]string{testNet0, testNet1, testNet2}))
+
+			for _, testNet := range []string{testNet0, testNet1, testNet2} {
+				podIFaceCacheDataResult, err := cache.ReadPodInterfaceCache(&baseCacheCreator, uid, testNet)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(podIFaceCacheDataResult.State).To(Equal(cache.PodIfaceNetworkPreparationFinished))
+			}
+		})
+
+		It("runs and fails at the config step, 2nd network", func() {
```

Comment:
IMO the description is confusing, how about: `should fail when processing the third face fails due to discovery failure`?

Reply:
Well, I think it is very clear without redundant words (e.g. `should`, `processing`).
Your suggestion is also incorrect: It is the 2nd failing (not the 3rd) and not at the discovery but at the config.